include/mega/node.h

```diff
@@ -337,7 +338,7 @@ struct MEGA_API LocalNode : public File
 
     void setnotseen(int);
 
-    void setfsid(handle);
+    void setfsid(handle newfsid, handlelocalnode_map& fsidnodes);
```

Comment:
I think we should not make this change - the old one is more efficient, and clearer.   We do always operate on the map from the client, and this signature implies we might operate on some other map.   Also it adds a lot of extra diff lines, for no functional change.

Reply:
This change is required because we cannot mock out the client. I doubt it's any less efficient with modern compilers.